### PR TITLE
Fix and improve loop example

### PR DESCRIPTION
**Pull request by mosabua**
## Description

Existing example had a logical and syntactical error. 

## Additional context and related issues

Thanks for finding that something is wrong @findepi 

## Release notes

(x) This is not user-visible or is docs only, and no release notes are required.

### CODE REVIEW

File: docs/src/main/sphinx/routines/loop.md [hunk 1]
```diff
@@ -24,27 +24,34 @@ block](routine-label).
...
-The following function counts up to `100` in a loop starting from the input
-value `i` and returns the number of incremental steps in the loop to get to
-`100`.
+The following function counts up to `100` with a step size `step` in a loop
+starting from the start value `i`, and returns the number of incremental steps
+in the loop to get to a value of `100` or higher:
 
 ```sql
-FUNCTION to_one_hundred(i int)
+FUNCTION to_one_hundred(i int, step int)
```
Comment (ebyhr) [inline docs/src/main/sphinx/routines/loop.md:32]: `start` might be better than `i`.

File: docs/src/main/sphinx/routines/loop.md [hunk 2]
```diff
@@ -24,27 +24,34 @@ block](routine-label).
...
+    RETURN count;
   END
 ```
 
+Example invocations:
+
+```sql
+SELECT to_one_hundred(90,1); --10
+SELECT to_one_hundred(0,5); --20
+SELECT to_one_hundred(12,3); -- 30
```
Comment (ebyhr) [inline docs/src/main/sphinx/routines/loop.md:52]: Most existing examples have a space among arguments:
```suggestion
SELECT to_one_hundred(90, 1); --10
SELECT to_one_hundred(0, 5); --20
SELECT to_one_hundred(12, 3); -- 30
```

**Review by ebyhr — APPROVED** [review]

File: docs/src/main/sphinx/routines/loop.md [hunk 3]
```diff
@@ -24,27 +24,34 @@ block](routine-label).
...
       IF i >= 100 THEN
         LEAVE abc;
-      END IF
+      END IF;
       SET count = count + 1;
-      SET i = i + 1;
+      SET i = i + step;
     END LOOP;
-    RETURN;
+    RETURN count;
```
Comment (findepi) [inline docs/src/main/sphinx/routines/loop.md:43]: now looks better for sure

File: docs/src/main/sphinx/routines/loop.md (hunk 2 above)
Comment (mosabua) [reply docs/src/main/sphinx/routines/loop.md:52]: Done

File: docs/src/main/sphinx/routines/loop.md (hunk 1 above)
Comment (mosabua) [reply docs/src/main/sphinx/routines/loop.md:32]: start is a keyword .. adapted to use `start_value` and then an internal value `current` .. and tested it all again. Good to go now.

**Pull request merged**